### ISSUE TITLE
perf(engine): pre-warm in-memory database by executing all transactions in parallel

### ISSUE THREAD

**Issue by rkrasiuk** [paradigmxyz/reth#12052]
## Description

Execute all transactions in parallel (perhaps if tx len or gas used exceeds certain threshold) to pre-warm in-memory state. Then execute the block with pre-warmed state.

https://github.com/paradigmxyz/reth/blob/2fba3c027b7396db868b162e16635197cd584822/crates/engine/tree/src/tree/mod.rs#L2174-L2180

## Additional Context

Related to https://github.com/paradigmxyz/reth/issues/11161 as it will allow us to start gathering proofs

**Comment by jenpaff** [comment]
feature wise this is already completed, will create new issue for enabling by default
